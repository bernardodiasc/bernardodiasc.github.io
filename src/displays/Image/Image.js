import React, { PureComponent } from 'react'
import './Image.css'

class Image extends PureComponent {
  static defaultProps = {
    alt: '',
    src: '',
    caption: '',
  }

  render() {
    const { alt, src, captions } = this.props
    return (
      <figure className="Image">
        <img alt={alt} src={src} />
        {captions && <figcaption>{captions}</figcaption>}
      </figure>
    )
  }
}

export default Image
