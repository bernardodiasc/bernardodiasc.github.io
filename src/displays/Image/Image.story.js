import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import StoryRouter from 'storybook-router'
import Provider from 'stories/containers/Provider'
import Docs from 'stories/containers/Docs'

import Image from './Image'
import ImageReadme from './README.md'
storiesOf('Displays/Image', module)
  .addDecorator(story => <Provider story={story()} />)
  .addDecorator(StoryRouter())
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{ImageReadme}</Docs>)
  .add('Example', () => <Image />)
  .add('Minimal render', () => <Image />)
