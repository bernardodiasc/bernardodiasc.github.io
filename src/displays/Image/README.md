# Image

## How to use

```
import Image from 'displays/Image'
```

```
<Image />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
