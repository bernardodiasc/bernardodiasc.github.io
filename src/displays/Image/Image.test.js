import React from 'react'
import { shallow } from 'enzyme'

import Image from './Image.js'

it('renders without crashing', () => {
  const component = shallow(<Image />)
  expect(component).toMatchSnapshot()
})
