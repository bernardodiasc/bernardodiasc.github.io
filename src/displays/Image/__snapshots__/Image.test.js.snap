// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<figure
  className="Image"
>
  <img
    alt=""
    src=""
  />
</figure>
`;
